types/package_ensure.pp

```diff
@@ -0,0 +1,2 @@
+# @summary Abstract for package_ensure values throughout the module
+type Mysql::Package_ensure = Variant[Enum['present','absent'], Pattern[/(\d+)[\.](\d+)[\.](\d+)/]]
```

Comment:
Is it worth it to re-use the `Stdlib::Ensure::Package` data type rather than introducing a new one?

puppetlabs/puppetlabs-stdlib/5633b0645699a82fbf5d02ac1cec0ca72c0e1c2e/types/ensure/package.pp#L2

Reply:
makes sense to me